runtime/services/catalog/artifacts/yaml/objects.go

```diff
@@ -425,6 +446,21 @@ func fromMetricsViewArtifact(metrics *MetricsView, path string) (*drivers.Catalo
 	}, nil
 }
 
+func validatedPolicyFieldList(fieldConditions []*ConditionalColumn, dimensions map[string]bool, property string) error {
+	if len(fieldConditions) > 0 {
+		for _, field := range fieldConditions {
+			if field.Name == "" || field.Condition == "" {
+				return fmt.Errorf("invalid 'policy': '%q' fields must have a valid name and condition", property)
+			}
+			// check the name is a valid dimension that exists in the metrics view
+			if !dimensions[field.Name] {
+				return fmt.Errorf("invalid 'policy': '%q' property [%q] does not exists in dimensions list", property, field.Name)
```

Comment:
The quoting is weird here – `'%q'` will output as `'"xxx"'`, and `[%q]` is inconsistent with other formatting. Maybe just use `%q` everywhere and nothing else around it?

Reply:
What about -
```
return fmt.Errorf("invalid 'policy': '%s' property [%s] does not exists in dimensions list", property, field.Name)
```

Reply:
so the key will be single quoted and value will be double quoted